File: src/com/nerdscorner/mvp/mvp/busevents/model/ModelComponent.java
package com.nerdscorner.mvp.mvp.busevents.model;

import java.io.File;

import com.nerdscorner.mvp.mvp.busevents.base.MvpComponent;

public class ModelComponent extends MvpComponent {
    private static final String MODEL_TEMPLATE = LANGUAGE_PLACEHOLDER + "/model_template";
    private static final String MODEL_JAVA = "Model." + LANGUAGE_EXTENSION_PLACEHOLDER;

    private static final String RELATIVE_PATH = File.separator + "ui" + File.separator + "mvp" + File.separator + "model";

    public ModelComponent(String basePath, String basePackage, String screenName, boolean shouldCreateWiring) {
        super(basePath, basePackage, screenName, MODEL_TEMPLATE, screenName + MODEL_JAVA, shouldCreateWiring);
    }

    @Override
    protected String relativePath() {
        return RELATIVE_PATH;
    }
}

File: src/com/nerdscorner/mvp/mvp/busevents/model/ModelComponent.kt
package com.nerdscorner.mvp.mvp.busevents.model

import com.nerdscorner.mvp.mvp.busevents.base.MvpComponent
import java.io.File

class ModelComponent(basePath: String, basePackage: String, screenName: String, shouldCreateWiring: Boolean)
    : MvpComponent(basePath, basePackage, screenName, MODEL_TEMPLATE, screenName + MODEL_JAVA, shouldCreateWiring) {

    override fun relativePath() = RELATIVE_PATH

    companion object {
        private const val MODEL_TEMPLATE = "$LANGUAGE_PLACEHOLDER/model_template"
        private const val MODEL_JAVA = "Model.$LANGUAGE_EXTENSION_PLACEHOLDER"

        private val RELATIVE_PATH = "${File.separator}ui${File.separator}mvp${File.separator}model"
    }
}
